src/utils.c

```diff
@@ -72,90 +72,86 @@ int b58_encode(const unsigned char *in, unsigned char length,
+
+uint8_t const BASE58_TABLE[] = {
+    0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF,  //
+    0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF,  //
+    0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF,  //
+    0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF,  //
+    0xFF, 0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0xFF, 0xFF,  //
+    0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0x09, 0x0A, 0x0B, 0x0C, 0x0D, 0x0E, 0x0F,  //
+    0x10, 0xFF, 0x11, 0x12, 0x13, 0x14, 0x15, 0xFF, 0x16, 0x17, 0x18, 0x19,  //
+    0x1A, 0x1B, 0x1C, 0x1D, 0x1E, 0x1F, 0x20, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF,  //
+    0xFF, 0x21, 0x22, 0x23, 0x24, 0x25, 0x26, 0x27, 0x28, 0x29, 0x2A, 0x2B,  //
+    0xFF, 0x2C, 0x2D, 0x2E, 0x2F, 0x30, 0x31, 0x32, 0x33, 0x34, 0x35, 0x36,  //
+    0x37, 0x38, 0x39, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF                           //
 };
-typedef uint64_t b58_maxint_t;
-typedef uint32_t b58_almostmaxint_t;
-#define b58_almostmaxint_bits (sizeof(b58_almostmaxint_t) * 8)
-static const b58_almostmaxint_t b58_almostmaxint_mask = ((((b58_maxint_t)1) << b58_almostmaxint_bits) - 1);
-bool b58_decode(void *bin, size_t *binszp, const char *b58, size_t b58sz)
-{
-	size_t binsz = *binszp;
-	const unsigned char *b58u = (void*)b58;
-	unsigned char *binu = bin;
-	size_t outisz = (binsz + sizeof(b58_almostmaxint_t) - 1) / sizeof(b58_almostmaxint_t);
-	b58_almostmaxint_t outi[outisz];
-	b58_maxint_t t;
-	b58_almostmaxint_t c;
-	size_t i, j;
-	uint8_t bytesleft = binsz % sizeof(b58_almostmaxint_t);
-	b58_almostmaxint_t zeromask = bytesleft ? (b58_almostmaxint_mask << (bytesleft * 8)) : 0;
-	unsigned zerocount = 0;
-
-	if (!b58sz)
-		b58sz = strlen(b58);
-
-	for (i = 0; i < outisz; ++i) {
-		outi[i] = 0;
-	}
-
-	// Leading zeros, just count
-	for (i = 0; i < b58sz && b58u[i] == '1'; ++i)
-		++zerocount;
-
-	for ( ; i < b58sz; ++i)	{
-		if (b58u[i] & 0x80)
-			// High-bit set on invalid digit
-			return false;
-		if (B58_DIGITS_MAP[b58u[i]] == -1)
-			// Invalid base58 digit
-			return false;
-		c = (unsigned)B58_DIGITS_MAP[b58u[i]];
-		for (j = outisz; j--; ) {
-			t = ((b58_maxint_t)outi[j]) * 58 + c;
-			c = t >> b58_almostmaxint_bits;
-			outi[j] = t & b58_almostmaxint_mask;
-		}
-		if (c)
-			// Output number too big (carry to the next int32)
-			return false;
-		if (outi[0] & zeromask)
-			// Output number too big (last int32 filled too far)
-			return false;
-	}
-
-	j = 0;
-	if (bytesleft) {
-		for (i = bytesleft; i > 0; --i) {
-			*(binu++) = (outi[0] >> (8 * (i - 1))) & 0xff;
-		}
-		++j;
-	}
-
-	for (; j < outisz; ++j)	{
-		for (i = sizeof(*outi); i > 0; --i) {
-			*(binu++) = (outi[j] >> (8 * (i - 1))) & 0xff;
-		}
-	}
-
-	// Count canonical base58 byte count
-	binu = bin;
-	for (i = 0; i < binsz; ++i)	{
-		if (binu[i])
-			break;
-		--*binszp;
-	}
-	*binszp += zerocount;
-
-	return true;
+
+int b58_decode(const char *in, size_t in_len, uint8_t *out, size_t out_len) {
```

Comment:
might used the version from crypto_helpers?

Reply:
I used the version from the standard app indeed